src/fty_common_socket_helpers.cc

```diff
@@ -0,0 +1,124 @@
+/*  =========================================================================
+    fty_common_socket_helpers - Helper functions for communication
+
+    Copyright (C) 2014 - 2019 Eaton
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+    =========================================================================
+*/
+
+/*
+@header
+    fty_common_socket_helpers - Helper functions for communication
+@discuss
+@end
+*/
+
+#include "fty_common_socket_helpers.h"
+
+
+#include <unistd.h>
+#include <stdexcept>
+
+#include <iostream>
+
+namespace fty
+{
+
+    Payload recvFrames(int socket)
+    {
+        //format => [ Number of frames ], [ <size of frame 1> <data> ], ... [ <size of frame N> <data> ]
+
+        //get the number of frames
+        uint32_t numberOfFrame = 0;
+
+        std::cerr << "reading data..." << std::endl;
+        
+        if(read(socket, &numberOfFrame, sizeof(uint32_t)) != sizeof(uint32_t))
```

Comment:
even if it is not relevant in our case, take care about the ENDIANNESS problem reading from/writing to socket.